compiler/rustc_hir_typeck/src/fn_ctxt/mod.rs

```diff
@@ -189,6 +189,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     pub fn errors_reported_since_creation(&self) -> bool {
         self.tcx.sess.err_count() > self.err_count_on_creation
     }
+
+    pub fn next_root_ty_var(&self, origin: TypeVariableOrigin) -> Ty<'tcx> {
```

Comment:
I put this in `FnCtxt` since I want to discourage it from being used in non-typeck places. I could also probably leave a FIXME to remove this once we model universes more faithfully. Also, I could give this a scarier name.

Reply:
think it's fine for this to not be too scary given that misusing it accepts strictly less code, would like some short comment about what this means though